Cabal/src/Distribution/Simple/Command.hs

```diff
@@ -651,7 +652,10 @@ commandsRun globalCommand commands args =
         _ -> do
           mCommand <- findExecutable $ "cabal-" <> name
           case mCommand of
-            Just exec -> callExternal flags exec cmdArgs
+            Just exec -> do
+                execPath <- getExecutablePath
+                void $ setEnv "CABAL" execPath
```

Comment:
It's probably fine, but this _does_ set a global var that lasts the lifetime of the process. As a paranoid programmer, I'd probably bracket the `callExternal` with a save/restore of the original value (if any).

Ideally we could exec() the new program on systems that support it, in which case this would be a moot point since "the lifetime of the process" would be rapidly drawing to a close. :)

Reply:
Fixed :)